src/ol/featureloader.js

```diff
@@ -87,8 +87,10 @@ ol.featureloader.tile = function(url, format) {
        * @this {ol.VectorTile}
        */
       function(features, dataProjection) {
-        this.setProjection(dataProjection);
-        this.setFeatures(features);
+        /** @type {ol.VectorTile} */
+        var me = this;
```

Comment:
I'd try the following:
```js
  /** 
   * @param {Array.<ol.Feature>} features The loaded features.
   * @param {ol.proj.Projection} dataProjection Data projection.
   * @this {ol.VectorTile}
   */
  function success(features, dataProjection) {
    this.setProjection(dataProjection);
    this.setFeatures(features);
  };  
  /** 
   * @this {ol.VectorTile}
   */
  function failure() {
    this.setState(ol.Tile.State.ERROR);
  };
```
The same (function without `var`, `@this` as last annotation) works everywhere else.

```